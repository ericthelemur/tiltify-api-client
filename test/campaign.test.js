const client = require("./test_client.js")


describe('campaign tests -', () => {

  beforeAll (async () => {
    await client.initialize()
  })

  test('correct Tiltify campaign returned', done => {
    function callback (data) {
      expect(data.amount_raised.value).toBe("425.00")
      done()
    }
    client.Campaigns.get('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  })

  test('correct Tiltify campaign recent donations returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    client.Campaigns.getRecentDonations('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  })

  test('correct Tiltify campaign donations returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    client.Campaigns.getDonations('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  }, 1000000)

  test('correct Tiltify campaign rewards returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    client.Campaigns.getRewards('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  })

  test('correct Tiltify campaign polls returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    client.Campaigns.getPolls('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  })

  test('correct Tiltify campaign targets returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    client.Campaigns.getTargets('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  })

  test('correct Tiltify campaign schedule returned', done => {
    function callback (data) {
      expect(data).toMatchSnapshot()
      done()
    }
    client.Campaigns.getSchedule('7099d7ba-cbce-40a6-8263-a5704d72b324', callback)
  })

})
