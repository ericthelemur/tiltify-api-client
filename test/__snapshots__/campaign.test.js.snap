// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`campaign tests - correct Tiltify campaign donations returned 1`] = `
[
  {
    "amount": {
      "currency": "GBP",
      "value": "10.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-12T21:06:58.747316Z",
    "created_at": "2024-12-12T21:06:27Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "Notorious LFG",
    "fundraising_event_id": null,
    "id": "cd1e74a9-4aff-4434-aed3-5e28df7e66fb",
    "is_match": false,
    "legacy_id": 8802858,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "6a313705-4b88-4458-a866-b74df61bcdba",
        "quantity": 1,
        "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
      },
    ],
    "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "50.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-12T21:06:13.169784Z",
    "created_at": "2024-12-12T21:05:41Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "overrrrit_",
    "fundraising_event_id": null,
    "id": "a4c1b23d-8f99-4b81-b9ae-2e68837df079",
    "is_match": false,
    "legacy_id": 8802856,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "f5588f17-386e-4cc3-8783-26b15705fc3d",
        "quantity": 1,
        "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
      },
    ],
    "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "250.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-09T22:33:48.169858Z",
    "created_at": "2024-12-04T22:33:49Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "rymbly",
    "fundraising_event_id": null,
    "id": "1158de30-580c-40ef-b406-3c748ac1252f",
    "is_match": true,
    "legacy_id": 8717516,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": null,
    "reward_id": null,
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "20.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-06T00:00:12.988484Z",
    "created_at": "2024-12-05T23:59:58Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": "I love food and I love when people give people food - sharing food is an act of love!",
    "donor_name": "MeganML",
    "fundraising_event_id": null,
    "id": "77363fc9-b847-4b9f-aa45-7da6e4c47537",
    "is_match": false,
    "legacy_id": 8724589,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "f5c1fa1d-c1e5-423b-b9d4-81f65019aced",
        "quantity": 1,
        "reward_id": "af4158c1-1183-4382-a1f3-7a0acf23cfe8",
      },
    ],
    "reward_id": "af4158c1-1183-4382-a1f3-7a0acf23cfe8",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "10.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T22:25:36.601290Z",
    "created_at": "2024-12-05T22:25:15Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": null,
    "donor_name": "poopoo",
    "fundraising_event_id": null,
    "id": "e802df7f-be37-47d0-aa04-d7db55e56789",
    "is_match": false,
    "legacy_id": 8723855,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "f4fb76ee-74b9-4791-b702-fad58a3ff75b",
        "quantity": 1,
        "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
      },
    ],
    "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "50.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T21:46:39.603526Z",
    "created_at": "2024-12-05T21:46:18Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": null,
    "donor_name": "rymbly",
    "fundraising_event_id": null,
    "id": "d306683a-3bea-4941-8398-f543b29006a3",
    "is_match": false,
    "legacy_id": 8723496,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "e37f9ab0-68ba-45b3-9098-8515e154aef5",
        "quantity": 1,
        "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
      },
    ],
    "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "20.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T21:06:07.199354Z",
    "created_at": "2024-12-05T21:05:35Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": "PAW Patrol, PAW Patrol
We'll be there on the double
Whenever there's a problem
'Round Adventure Bay
Ryder and his team of pups
Will come and save the day
Marshall, Rubble, Chase
Rocky, Zuma, Skye
Yeah, they're on the way
PAW Patrol, PAW Patrol",
    "donor_name": "bushhhyy",
    "fundraising_event_id": null,
    "id": "bdb957aa-f8e4-4d07-97f0-061083c7784e",
    "is_match": false,
    "legacy_id": 8723139,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "25767ac6-c190-416f-8677-f3296d500758",
        "quantity": 1,
        "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
      },
    ],
    "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "5.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T17:23:05.626303Z",
    "created_at": "2024-12-05T17:21:55Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": ":>",
    "donor_name": "Secret",
    "fundraising_event_id": null,
    "id": "ad6fd3e8-f022-43ac-b8f0-21a9b4a70a1e",
    "is_match": false,
    "legacy_id": 8721220,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "d6ddfdd8-f205-4a8c-af91-58816014cd70",
        "quantity": 1,
        "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
      },
    ],
    "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "10.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-11-25T22:18:15.283882Z",
    "created_at": "2024-11-25T22:18:06Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "rachiewease",
    "fundraising_event_id": null,
    "id": "07f3e005-7928-4447-afa2-8e03ee5225f2",
    "is_match": false,
    "legacy_id": 8663722,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": null,
    "reward_id": null,
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
]
`;

exports[`campaign tests - correct Tiltify campaign polls returned 1`] = `[]`;

exports[`campaign tests - correct Tiltify campaign recent donations returned 1`] = `
[
  {
    "amount": {
      "currency": "GBP",
      "value": "10.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-12T21:06:58.747316Z",
    "created_at": "2024-12-12T21:06:27Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "Notorious LFG",
    "fundraising_event_id": null,
    "id": "cd1e74a9-4aff-4434-aed3-5e28df7e66fb",
    "is_match": false,
    "legacy_id": 8802858,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "6a313705-4b88-4458-a866-b74df61bcdba",
        "quantity": 1,
        "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
      },
    ],
    "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "50.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-12T21:06:13.169784Z",
    "created_at": "2024-12-12T21:05:41Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "overrrrit_",
    "fundraising_event_id": null,
    "id": "a4c1b23d-8f99-4b81-b9ae-2e68837df079",
    "is_match": false,
    "legacy_id": 8802856,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "f5588f17-386e-4cc3-8783-26b15705fc3d",
        "quantity": 1,
        "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
      },
    ],
    "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "250.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-09T22:33:48.169858Z",
    "created_at": "2024-12-04T22:33:49Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "rymbly",
    "fundraising_event_id": null,
    "id": "1158de30-580c-40ef-b406-3c748ac1252f",
    "is_match": true,
    "legacy_id": 8717516,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": null,
    "reward_id": null,
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "20.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-06T00:00:12.988484Z",
    "created_at": "2024-12-05T23:59:58Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": "I love food and I love when people give people food - sharing food is an act of love!",
    "donor_name": "MeganML",
    "fundraising_event_id": null,
    "id": "77363fc9-b847-4b9f-aa45-7da6e4c47537",
    "is_match": false,
    "legacy_id": 8724589,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "f5c1fa1d-c1e5-423b-b9d4-81f65019aced",
        "quantity": 1,
        "reward_id": "af4158c1-1183-4382-a1f3-7a0acf23cfe8",
      },
    ],
    "reward_id": "af4158c1-1183-4382-a1f3-7a0acf23cfe8",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "10.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T22:25:36.601290Z",
    "created_at": "2024-12-05T22:25:15Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": null,
    "donor_name": "poopoo",
    "fundraising_event_id": null,
    "id": "e802df7f-be37-47d0-aa04-d7db55e56789",
    "is_match": false,
    "legacy_id": 8723855,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "f4fb76ee-74b9-4791-b702-fad58a3ff75b",
        "quantity": 1,
        "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
      },
    ],
    "reward_id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "50.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T21:46:39.603526Z",
    "created_at": "2024-12-05T21:46:18Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": null,
    "donor_name": "rymbly",
    "fundraising_event_id": null,
    "id": "d306683a-3bea-4941-8398-f543b29006a3",
    "is_match": false,
    "legacy_id": 8723496,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "e37f9ab0-68ba-45b3-9098-8515e154aef5",
        "quantity": 1,
        "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
      },
    ],
    "reward_id": "46b4fe7c-f861-4fed-960a-b1df13457175",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "20.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T21:06:07.199354Z",
    "created_at": "2024-12-05T21:05:35Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": "PAW Patrol, PAW Patrol
We'll be there on the double
Whenever there's a problem
'Round Adventure Bay
Ryder and his team of pups
Will come and save the day
Marshall, Rubble, Chase
Rocky, Zuma, Skye
Yeah, they're on the way
PAW Patrol, PAW Patrol",
    "donor_name": "bushhhyy",
    "fundraising_event_id": null,
    "id": "bdb957aa-f8e4-4d07-97f0-061083c7784e",
    "is_match": false,
    "legacy_id": 8723139,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "25767ac6-c190-416f-8677-f3296d500758",
        "quantity": 1,
        "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
      },
    ],
    "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "5.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-12-05T17:23:05.626303Z",
    "created_at": "2024-12-05T17:21:55Z",
    "donation_matches": [
      {
        "active": false,
        "amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "completed_at": "2024-12-09T22:33:48Z",
        "donation_id": "1158de30-580c-40ef-b406-3c748ac1252f",
        "ends_at": "2024-12-09T22:33:47Z",
        "id": "8d87a5cd-757a-48dc-8e1e-db82df40db16",
        "inserted_at": "2024-12-04T22:33:51Z",
        "matched_by": "rymbly",
        "pledged_amount": {
          "currency": "GBP",
          "value": "250.00",
        },
        "started_at_amount": {
          "currency": "GBP",
          "value": "10.00",
        },
        "starts_at": "2024-12-09T22:33:47Z",
        "total_amount_raised": {
          "currency": "GBP",
          "value": "105.00",
        },
        "updated_at": "2024-12-09T22:33:48Z",
      },
    ],
    "donor_comment": ":>",
    "donor_name": "Secret",
    "fundraising_event_id": null,
    "id": "ad6fd3e8-f022-43ac-b8f0-21a9b4a70a1e",
    "is_match": false,
    "legacy_id": 8721220,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": [
      {
        "id": "d6ddfdd8-f205-4a8c-af91-58816014cd70",
        "quantity": 1,
        "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
      },
    ],
    "reward_id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
  {
    "amount": {
      "currency": "GBP",
      "value": "10.00",
    },
    "campaign_id": "7099d7ba-cbce-40a6-8263-a5704d72b324",
    "cause_id": "802d8e02-3498-4be2-bfeb-a0036ec98e78",
    "completed_at": "2024-11-25T22:18:15.283882Z",
    "created_at": "2024-11-25T22:18:06Z",
    "donation_matches": null,
    "donor_comment": null,
    "donor_name": "rachiewease",
    "fundraising_event_id": null,
    "id": "07f3e005-7928-4447-afa2-8e03ee5225f2",
    "is_match": false,
    "legacy_id": 8663722,
    "poll_id": null,
    "poll_option_id": null,
    "reward_claims": null,
    "reward_id": null,
    "sustained": false,
    "target_id": null,
    "team_event_id": null,
  },
]
`;

exports[`campaign tests - correct Tiltify campaign rewards returned 1`] = `
[
  {
    "active": true,
    "amount": {
      "currency": "GBP",
      "value": "5",
    },
    "description": "All who enter this will be put in a wheel spin raffle draw (only 1 redeem per person will count) Whoever’s name is chosen and announced winner (1 winner only) will have a VIP badge in my stream for an entire year! Subscribers to my channel who redeem this get an additional entry to the raffle!",
    "ends_at": null,
    "fair_market_value": null,
    "highlighted": false,
    "id": "16c4b351-080f-41d7-baa1-8630b08f11cf",
    "image": {
      "alt": "alt",
      "height": 600,
      "src": "https://assets.tiltify.com/assets/default-reward.png",
      "width": 600,
    },
    "inserted_at": "2024-11-22T04:22:09Z",
    "legacy_id": 262593,
    "name": "VIP raffle entry",
    "quantity": null,
    "quantity_remaining": null,
    "starts_at": null,
    "updated_at": "2024-11-25T19:32:59Z",
  },
  {
    "active": true,
    "amount": {
      "currency": "GBP",
      "value": "10",
    },
    "description": "I’ll draw anything of your choice (within reason) on my ipad! All drawings will be live streamed in my discord :)",
    "ends_at": null,
    "fair_market_value": null,
    "highlighted": false,
    "id": "b9a8292c-c4b9-47e4-a4fb-c1ba5df9c700",
    "image": {
      "alt": "alt",
      "height": 600,
      "src": "https://assets.tiltify.com/assets/default-reward.png",
      "width": 600,
    },
    "inserted_at": "2024-11-09T04:02:14Z",
    "legacy_id": 261108,
    "name": "Draw something on Procreate",
    "quantity": 10,
    "quantity_remaining": 8,
    "starts_at": null,
    "updated_at": "2024-11-25T19:33:08Z",
  },
  {
    "active": true,
    "amount": {
      "currency": "GBP",
      "value": "20",
    },
    "description": "Gain either a custom discord role for my server or your very own text command for my twitch channel, choice is yours!",
    "ends_at": null,
    "fair_market_value": null,
    "highlighted": false,
    "id": "af4158c1-1183-4382-a1f3-7a0acf23cfe8",
    "image": {
      "alt": "alt",
      "height": 600,
      "src": "https://assets.tiltify.com/assets/default-reward.png",
      "width": 600,
    },
    "inserted_at": "2024-11-09T04:03:16Z",
    "legacy_id": 261109,
    "name": "Custom discord role or chat command",
    "quantity": 5,
    "quantity_remaining": 4,
    "starts_at": null,
    "updated_at": "2024-11-25T19:33:16Z",
  },
  {
    "active": true,
    "amount": {
      "currency": "GBP",
      "value": "50",
    },
    "description": "Give me a single player game (must be available on Playstation 4 or 5) to add to my streaming backlog. This will be a game I will have to stream on Twitch when there’s a gap in the schedule :)",
    "ends_at": null,
    "fair_market_value": null,
    "highlighted": false,
    "id": "46b4fe7c-f861-4fed-960a-b1df13457175",
    "image": {
      "alt": "alt",
      "height": 600,
      "src": "https://assets.tiltify.com/assets/default-reward.png",
      "width": 600,
    },
    "inserted_at": "2024-11-09T04:04:41Z",
    "legacy_id": 261110,
    "name": "Add a single player game to my backlog",
    "quantity": 3,
    "quantity_remaining": 1,
    "starts_at": null,
    "updated_at": "2024-11-25T19:33:26Z",
  },
]
`;

exports[`campaign tests - correct Tiltify campaign schedule returned 1`] = `
[
  {
    "description": null,
    "ends_at": "2024-12-15T20:00:00Z",
    "id": "d31405c5-6ba6-48a6-a597-b9f348fda7df",
    "inserted_at": "2024-11-09T03:48:49Z",
    "legacy_id": 130257,
    "name": "Campaign runtime",
    "starts_at": "2024-11-25T20:00:00Z",
    "updated_at": "2024-11-09T03:48:49Z",
  },
]
`;

exports[`campaign tests - correct Tiltify campaign targets returned 1`] = `[]`;
